### PR TITLE
Add unit tests for OrderModel

### PR DESCRIPTION
**Pull request by kiran-mastek**
Story: [AB#13236](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/13236)
Task: [AB#13949](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/13949)

### CODE REVIEW

**Review by bjss-chris — COMMENTED** [review]
Rather than update `OrderCustomization` to create order items by default I think it might be better to introduce a new `OrderItemsCustomization` (or something similar). That way we can make it opt-in, either by introducing a new auto data attribute or by parameterizing the existing attribute. We would also avoid having to explicitly remove order items in the tests that don't need them.

What do you think?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderItemsControllerTests.cs [hunk 1]
```diff
@@ -23,6 +23,7 @@
 using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.AutoFixture;
 using NHSD.BuyingCatalogue.Ordering.Api.Validation;
 using NHSD.BuyingCatalogue.Ordering.Common.Constants;
+using NHSD.BuyingCatalogue.Ordering.Common.UnitTests;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderItemsControllerTests.cs:26]: Still needed?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/ServiceRecipientsSectionControllerTests.cs [hunk 2]
```diff
@@ -15,6 +15,7 @@
 using NHSD.BuyingCatalogue.Ordering.Api.Models;
 using NHSD.BuyingCatalogue.Ordering.Api.Services;
 using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.AutoFixture;
+using NHSD.BuyingCatalogue.Ordering.Common.UnitTests;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/ServiceRecipientsSectionControllerTests.cs:18]: Still needed?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Models/OrderModelTests.cs [hunk 3]
```diff
@@ -0,0 +1,366 @@
+﻿using System;
+using System.Linq;
+using FluentAssertions;
+using NHSD.BuyingCatalogue.Ordering.Api.Models;
+using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.AutoFixture;
+using NHSD.BuyingCatalogue.Ordering.Common.UnitTests.AutoFixture;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Models/OrderModelTests.cs:6]: Do we still need `System.Linq` and `NHSD.BuyingCatalogue.Ordering.Common.UnitTests.AutoFixture`?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/CompleteOrderServiceTests.cs [hunk 4]
```diff
@@ -12,6 +12,7 @@
 using NHSD.BuyingCatalogue.Ordering.Api.Services.CompleteOrder;
 using NHSD.BuyingCatalogue.Ordering.Api.Services.CreatePurchasingDocument;
 using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.AutoFixture;
+using NHSD.BuyingCatalogue.Ordering.Common.UnitTests;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/CompleteOrderServiceTests.cs:15]: Still needed?

File: tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs [hunk 5]
```diff
@@ -5,6 +5,7 @@
 using AutoFixture.NUnit3;
 using FluentAssertions;
 using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.AutoFixture;
+using NHSD.BuyingCatalogue.Ordering.Common.UnitTests;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs:8]: Still needed?

**Review by bjss-chris — CHANGES_REQUESTED** [review]
I think there might be some redundant `using` statements but otherwise looks good.

File: tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs (hunk 5 above)
Comment (kiran-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs:8]: removed

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/CompleteOrderServiceTests.cs (hunk 4 above)
Comment (kiran-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/CompleteOrderServiceTests.cs:15]: removed

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Models/OrderModelTests.cs (hunk 3 above)
Comment (kiran-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Models/OrderModelTests.cs:6]: removed

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/ServiceRecipientsSectionControllerTests.cs (hunk 2 above)
Comment (kiran-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/ServiceRecipientsSectionControllerTests.cs:18]: removed

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderItemsControllerTests.cs (hunk 1 above)
Comment (kiran-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderItemsControllerTests.cs:26]: removed

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**